src/Import.ts

```diff
@@ -387,5 +387,8 @@ export function maybeRelativePath(outputPath: string, importPath: string): strin
 
 /** Checks if `path1 === path2` despite minor path differences like `./foo` and `foo`. */
 export function sameModule(path1: string, path2: string): boolean {
-  return path1 === path2 || path.resolve(path1) === path.resolve(path2);
+  // TypeScript: import paths ending in .js and .ts are resolved to the .ts file.
+  const basePath1 = path1.replace(/\.[tj]sx?/, '');
+  const basePath2 = path2.replace(/\.[tj]sx?/, '');
+  return basePath1 === basePath2 || path.resolve(path1) === path.resolve(path2);
```

Comment:
@paralin should the `path.resolve(path1)` use the `basePath1` as well? Afaiu `path.resolve` doesn't check actual file existence, it just does the path manipulation logic, so should be fine?

Reply:
@stephenh done